sklearn/inspection/_partial_dependence.py

```diff
@@ -172,6 +179,95 @@ def _partial_dependence_brute(est, grid, features, X, response_method):
     return averaged_predictions
 
 
+def _validate_parameters(estimator, X, features, response_method, method):
```

Comment:
Let's move the validation on the top of the file. Then _grid_from_X and the the different _partial_dependence and _ice_brute

Reply:
Moved in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/640f938124fd8442fdee4c3e32b0cb2cc108fbbc